projects/wavelet_dataAggregation/checkDataType.py

```diff
@@ -0,0 +1,188 @@
+  # meanSamplingInterval = np.mean(samplingIntervals)
+  # samplingIntervalsStd = np.std(samplingIntervals)
+  # return meanSamplingInterval, samplingIntervalsStd
+
+
+def checkDataType(dataFilePath):
+  """
+  Return the data type (transactional or non-transactional)
+  The data type is determined via an data type indicator, defined as the
+  ratio between median absolute deviation and median of the sampling interval.
+
+  @param dataFilePath: Name of a csv file, the file must have two columns
+  with header "timestamp", and "value"
+  @return dataType: a string with value "transactional" or "non-transactional"
+  """
+  (timestamp, sig) = readCSVfiles(dataFilePath)
+  medianSamplingInterval, medianAbsoluteDev = estimateMedianAbsoluteDeviation(timestamp)
+
+  dataTypeIndicator = medianAbsoluteDev/medianSamplingInterval
+  if dataTypeIndicator > 0.2:
+    dataType = "transactional"
+  else:
+    dataType = "non-transactional"
+
+  return dataType, dataTypeIndicator
+
+
+def checkAlgorithmPerformance(dataTypeList, predictedDataTypeList, dataTypeIndicatorList):
+  """
+  Check the performance of the data type detection algorithm
+  :param dataTypeList: List of data type
+  :param predictedDataTypeList: List of predicted data type
+  :param dataTypeIndicator: List of the datatype indicator
+  """
+
+  numCorrect = 0
+  for i in xrange(len(dataTypeList)):
+    if dataTypeList[i] == predictedDataTypeList[i]:
+      numCorrect += 1
+
+  correctRate = float(numCorrect)/len(dataTypeList)
+  print "correct rate: ", str(correctRate)
+
+
+if __name__ == "__main__":
+
+  print "Test algorithm on river-view data streams ..."
+  fileNameList, filePathList, dataTypeList = downloadExampleData()
+  dataTypeIndicatorList = []
+  predictedDataTypeList = []
+  for i in xrange(len(filePathList)):
+    filePath = filePathList[i]
+    predictedDataType, dataTypeIndicator = checkDataType(filePath)
+    print " File: ", fileNameList[i], " DataType: ", dataTypeList[i], \
+          " predicted Data Type: ", predictedDataType
+    predictedDataTypeList.append(predictedDataType)
+    dataTypeIndicatorList.append(dataTypeIndicator)
+  checkAlgorithmPerformance(dataTypeList, predictedDataTypeList, dataTypeIndicatorList)
+
+
+  print "Test algorithm on NAB data ..."
+  NABPath = '/Users/ycui/nta/NAB/'
+  dataPath=NABPath+'data/'
+  dataDirs = [join(dataPath, f) for f in listdir(dataPath) if not isfile(join(dataPath, f))]
+
+  dataTypeList = []
+  predictedDataTypeList = []
+  dataTypeIndicatorList = []
+  for dir in dataDirs:
+    dataFileList = [join(dir, f) for f in listdir(dir) if isfile(join(dir, f))]
+
+    for dataFile in dataFileList:
+      predictedDataType, dataTypeIndicator = checkDataType(dataFile)
+
+      predictedDataTypeList.append(predictedDataType)
+      dataTypeIndicatorList.append(dataTypeIndicator)
+      dataTypeList.append('non-transactional')
+
+      print " File: ", dataFile[i], " DataType: ", 'non-transactional', \
```

Comment:
There is something weird with this `i` variable. It's the one from `for i in xrange(len(filePathList)):` and is always equals to 7 here. @ywcui1990 can you look at this again and check that the 100% accuracy result is actually correct? thanks